exercises/pascals-triangle/src/example.c

```diff
@@ -0,0 +1,26 @@
+#include "pascals_triangle.h"
+
+long long unsigned int factorial(int n)
+{
+   return (n == 0) ? 1 : n * factorial(n - 1);
+}
+
+long long unsigned int combination(int n, int r)
+{
+   return (r > n) ? 0 : factorial(n) / (factorial(n - r) * factorial(r));
+}
+
+int pascals_triangle(int count, int input[][((count < 0) ? 1 : count)])
+{
+   if (count < 0) {
+      memset(input, -1, sizeof(int) * count * count);
+      return -1;
+   }
+   memset(input, 0, sizeof(int) * count * count);
```

Comment:
When count is zero, this should result in a warning. It also means that when count is zero the value in input is left at whatever it was previously - it is never actually explicitly set to 0. the test only passes because your environment happens to set the `auto` scoped `result` array to all zeros by default.